index.js

```diff
@@ -319,6 +319,7 @@ Swarm.prototype._onconnection = function (connection, type, peer) {
   function onhandshake (remoteId) {
     if (!remoteId) remoteId = connection.remoteId
     clearTimeout(timeout)
+    if (connection.id) idHex = connection.id.toString('hex')
```

Comment:
could you move this into the if (this._stream) { } section above? then there are no surprises if tcp/utp accidentally adds an id field in the future.